webhook/resourcesemantics/defaulting/defaulting_test.go

```diff
@@ -323,6 +353,72 @@ func TestAdmitCreates(t *testing.T) {
 			Path:      "/metadata/annotations/pkg.knative.dev~1creator",
 			Value:     user1,
 		}},
+	}, {
+		name: "test simple creation (callback return error)",
+		setup: func(ctx context.Context, resource *Resource) {
+			resource.Spec.FieldForCallbackDefaulting = "no magic value"
+		},
+		rejection: "no magic value",
+	}, {
+		name: "test simple creation (resource and callback defaults)",
+		setup: func(ctx context.Context, r *Resource) {
+			r.SetDefaults(ctx)
+			r.Spec.FieldForCallbackDefaulting = "magic value"
+			// THIS IS NOT WHO IS CREATING IT, IT IS LIES!
+			r.Annotations = map[string]string{
+				"pkg.knative.dev/lastModifier": user2,
+			}
+		},
+		patches: []jsonpatch.JsonPatchOperation{{
+			Operation: "replace",
+			Path:      "/spec/fieldDefaultingCallback",
+			Value:     "I'm a default",
+		}, {
+			Operation: "replace",
+			Path:      "/metadata/annotations/pkg.knative.dev~1lastModifier",
+			Value:     user1,
+		}, {
+			Operation: "add",
+			Path:      "/metadata/annotations/pkg.knative.dev~1creator",
+			Value:     user1,
+		}, {
+			Operation: "add",
+			Path:      "/spec/FieldForCallbackDefaultingUsername",
+			Value:     user1,
+		}},
+	}, {
+		name: "test simple creation (only callback defaults)",
+		setup: func(ctx context.Context, r *Resource) {
+			r.TypeMeta.APIVersion = "pkg.knative.dev/v1beta1"
+			r.TypeMeta.Kind = "ResourceCallbackDefault"
+			r.Spec.FieldForCallbackDefaulting = "magic value"
+			// THIS IS NOT WHO IS CREATING IT, IT LIES!
+			r.Annotations = map[string]string{
+				"pkg.knative.dev/lastModifier": user2,
+			}
```

Comment:
Should this be set given that the resource is being 'created' and not 'updated'?

Reply:
if I understand your question, it was set 
```golang
{
			Operation: "replace",
			Path:      "/spec/fieldDefaultingCallback",
			Value:     "I'm a default",
}
```

The problem is that the json tag was called differently from the field name of the struct.
I pushed a new commit to fix it.